xml-csharp/SelectingXmlNodesWithXpath/SelectingXmlNodesWithXpathTests/SelectingXmlNodesWithXpathTest.cs

```diff
@@ -0,0 +1,99 @@
+using System.Xml;
+
+namespace SelectingXmlNodesWithXpathTests;
+
+public class SelectingXmlNodesWithXpathTest

```

Comment:
The unit tests should be testing the code that we are adding to the article, so the class under test here should be the `XmlNodeSelector`. We don't need to write unit tests to validate the framework. I should have caught that in my last review, my bad.

Reply:
I totally agree with you about how the tests must be made. For this project, I looked at other projects in the code-maze repository for reference and they are using the initial testing implementation.

I removed the print from inside the methods and changed their returning types from void to `string` and `List<string>`. This way, the data printing was passed to the `Program.cs`.

I liked more the actual result. I will update the article to reflect the changes.